src/main/java/Duke.java

```diff
@@ -1,10 +1,143 @@
+import java.util.*;
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
+    private Task task;
+
+    public static void main(String[] args) throws DukeException {
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+        Duke newDuke = new Duke();
+        newDuke.chatDuke();
+
+
+        /*String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello from\n" + logo);*/
     }
+
+    public void chatDuke() throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        ArrayList<Task> taskArray = new ArrayList<>();
+        String str = sc.nextLine();
+        int counter = 0;
+        if (str.equals("bye")) {
+            System.out.println("Bye. Hope to see you again soon");
+        }
+        while (!str.equals("bye")){
+            if (str.equals("list")) {
+                returnArrayStrings(taskArray);
+                str = sc.nextLine();
+            }
+
+            else if (str.equals("mark")) {
+                int num = sc.nextInt();
+                Task currTask = taskArray.get(num - 1);
+                currTask.mark();
+                System.out.println("Nice! I've marked this task as done:\n" +
+                        "[" + currTask.getStatusIcon() + "] " + currTask.description);
+                str = sc.nextLine();
+                str = sc.nextLine();
+            }
+
+            else if (str.equals("unmark")) {
+                int num = sc.nextInt();
+                Task currTask = taskArray.get(num - 1);
+                currTask.unMark();
+                System.out.println("OK, I've marked this task as not done yet:\n" +
+                        "[" + currTask.getStatusIcon() + "] " + currTask.description);
+                str = sc.nextLine();
+                str = sc.nextLine();
+            }
+            else if (str.equals("todo")) {
+                String input = sc.nextLine();
+                checkInput(input);
+                taskArray.add(new ToDo(input));
+                System.out.println("Got it. I've added this task:\n" + "  " +
+                        taskArray.get(counter));
+                counter++;
+                System.out.println("Now you have " + counter + " tasks in the list.");
+                str = sc.nextLine();
+            }
+            else if (str.equals("deadline")) {
+                String input = sc.nextLine();
+                checkInput(input);
+                String by = sc.nextLine();
+                checkInput(by);
```

Comment:
It also seems like there is no indicator for the user to know what to enter at which point, since your arguments are all separated by a newline. Maybe some feedback would be good for UX?